zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Schema.scala

```diff
@@ -39,42 +39,93 @@ object Relation extends Enumeration {
   val ~= = Value("~=")
 }
 
-class Field(val name: String, val dataType: DataType, val isOptional: Boolean = false) {
+trait Field {
+  val name: String
+  val dataType: DataType
+  val isOptional: Boolean = false
+
+  def as(name: String): Field
 }
 
-case class NumberField(override val name: String, override val isOptional: Boolean = false)
-  extends Field(name, DataType.Number, isOptional)
+object Field {
+  def apply(name: String, dataType: DataType, isOptional: Boolean = false): Field = {
+    dataType match {
+      case DataType.Number => NumberField(name, isOptional)
+      case DataType.Time => TimeField(name, isOptional)
+      case DataType.String => StringField(name, isOptional)
+      case DataType.Text => TextField(name, isOptional)
+      case DataType.Point => PointField(name, isOptional)
+      case DataType.Boolean => PointField(name, isOptional)
+      case _ => ???
+    }
+  }
+
+}
+
+case class NumberField(override val name: String, override val isOptional: Boolean = false) extends Field {
+  override val dataType = DataType.Number
+
+  def as(name: String) = NumberField(name, isOptional)
```

Comment:
now I see the usage of it. but I'm still not convinced this should be a functionality of a `Field`. 
If this function helps a lot, we could add an object `as` function just like the `apply` in the `object Field`.